Spokestack/AppleSpeechRecognizer.swift

```diff
@@ -69,6 +72,8 @@ import Speech
     
     private func activate() {
         do {
+            self.configuration.tracing.rawValue <= Trace.Level.DEBUG.rawValue  ?
```

Comment:
So if tracing's set to debug, the audio engine won't actually start, but just dump info? Trying to parse the ternary here...this could stand another pass for readability.

Reply:
the else part of the ternary is empty. the switch is only there because getting the debug information is costly so we don't want to do it in prod, and it's a ternary to make it all a one-liner so that it doesn't get in the way of the main logic.

Reply:
Whew, OK. I like to encapsulate the level checks in another class, though that gets messy in its own way.

I think just having an if statement would be less head-scratchy for readers—or you could even make it a separate fn with no extra fuss since it's not using any locals to dump the info. That might be the minimum distraction from the main logic.

Also: you prefix everything but `bufferSize` with `self.`.

Reply:
the level checks are done in another class, but in this case the trace message information is costly so a duplicate check beforehand makes sense.